otj-server-mvc/src/main/java/com/opentable/server/mvc/LoggingHandlerExceptionResolver.java

```diff
@@ -0,0 +1,105 @@
+package com.opentable.server.mvc;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.ConversionNotSupportedException;
+import org.springframework.beans.TypeMismatchException;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.core.Ordered;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.http.converter.HttpMessageNotWritableException;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindException;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+import org.springframework.web.HttpMediaTypeNotSupportedException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.MissingPathVariableException;
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.bind.ServletRequestBindingException;
+import org.springframework.web.context.request.async.AsyncRequestTimeoutException;
+import org.springframework.web.multipart.support.MissingServletRequestPartException;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+@Component
+@Conditional(LoggingHandlerExceptionResolver.InstallLoggingHandlerExceptionResolver.class)
+public class LoggingHandlerExceptionResolver implements HandlerExceptionResolver, Ordered {
+
+    public static class InstallLoggingHandlerExceptionResolver implements Condition {
+        @Override
+        public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+            final String value = context.getEnvironment().
+                getProperty("ot.server.mvc.log-resolved-exception", "true");
+            return "true".equals(value);
+        }
+    }
+
+    private static final Logger log = LoggerFactory.getLogger(LoggingHandlerExceptionResolver.class);
+
+    @Override
+    public int getOrder() {
+        return Ordered.HIGHEST_PRECEDENCE;
+    }
+
+    /**
+     * Log stack trace for all exceptions which going to be resolved by
+     * {@link org.springframework.web.servlet.mvc.support.DefaultHandlerExceptionResolver}
+     */
+    @Override
+    public ModelAndView resolveException(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) {
+        if (ex instanceof HttpRequestMethodNotSupportedException) {
```

Comment:
How about building a static immutableSet and doing a contains!

Reply:
Because right now this is kind of ugly and hard to maintain. Can you explain your choices and their hierarchy?